### PR TITLE
Basic logging of what's happening inside Raven

### PR DESCRIPTION
**Pull request by pswierad-src**
Added basic logging to make raven work more clear. Expanded on the examples to account for those changes.

### CODE REVIEW

File: Demos/Raven.NET.Demo.Console/Program.cs [hunk 1]
```diff
@@ -21,12 +21,13 @@ static void Main(string[] args)
                 .ConfigureServices((_, services) =>
                 {
                     services.ConfigureRaven(builder.Build());
```
Comment (wisedev-code) [inline Demos/Raven.NET.Demo.Console/Program.cs:23]: Maybe all of those registration can be part of Configure method? Now got this idea

File: Raven.NET.Core/Observers/RavenTypeWatcher.cs [hunk 2]
```diff
@@ -73,31 +90,35 @@ public IRavenTypeWatcher Create<T>(string name, string keyName, Func<RavenSubjec
...
         /// <inheritdoc/>
         public void Stop(string name)
         {
+            _logger.LogDebug($"Stopping raven {name}");
             var raven = _ravenProvider.GetRaven(name) as RavenTypeWatcher;
             raven._watchedSubjects.ForEach(x => x.Detach(this));
+            _logger.LogInformation($"Detached {raven._watchedSubjects.Count} subjects from raven {RavenName}");
 
             if (raven._ravenSettings.AutoDestroy)
-            {
```
Comment (wisedev-code) [inline Raven.NET.Core/Observers/RavenTypeWatcher.cs:85]: Why brackets removed?

File: Raven.NET.Core/Observers/RavenWatcher.cs [hunk 3]
```diff
@@ -64,27 +85,35 @@ public void UnWatch(string name, RavenSubject subject)
         {
             var raven = _ravenProvider.GetRaven(name) as RavenWatcher;
             raven.watchedSubjects.Remove(subject);
-            
-            if(raven._ravenSettings.AutoDestroy)
+            _logger.LogInformation($"Subject {subject.UniqueId} is no longer watched by {RavenName}");
+
+            if (!raven._ravenSettings.AutoDestroy)
```
Comment (wisedev-code) [inline Raven.NET.Core/Observers/RavenWatcher.cs:90]: Add brackets, I think we discussed to always have those, correct me if i am wrong

File: Raven.NET.Core/Subjects/RavenSubject.cs [hunk 4]
```diff
@@ -14,7 +14,7 @@ namespace Raven.NET.Core.Subjects
     public class RavenSubject
     {
         internal List<IRaven> Observers = new();
-        private Guid UniqueId { get; set; }
+        internal Guid UniqueId { get; set; }
```
Comment (wisedev-code) [inline Raven.NET.Core/Subjects/RavenSubject.cs:17]: Why this change needed?

**Review by ghost — COMMENTED** [review]
Just a few comments

File: Raven.NET.Core/Observers/RavenWatcher.cs (hunk 3 above)
Comment (pswierad-src) [reply Raven.NET.Core/Observers/RavenWatcher.cs:90]: True, will change

File: Raven.NET.Core/Observers/RavenTypeWatcher.cs (hunk 2 above)
Comment (pswierad-src) [reply Raven.NET.Core/Observers/RavenTypeWatcher.cs:85]: They should be there

File: Demos/Raven.NET.Demo.Console/Program.cs (hunk 1 above)
Comment (pswierad-src) [reply Demos/Raven.NET.Demo.Console/Program.cs:23]: good idea. We could have just one extension method "AddRaven()" that will do all this work

File: Raven.NET.Core/Subjects/RavenSubject.cs (hunk 4 above)
Comment (pswierad-src) [reply Raven.NET.Core/Subjects/RavenSubject.cs:17]: Thought some logs may use having Id logged. In case something goes wrong we would know in which subject the problem appeared. If that's not necessary I'll remove it. I'm fine both ways

Comment (wisedev-code) [reply Raven.NET.Core/Subjects/RavenSubject.cs:17]: Nah its fine

**Review by ghost — APPROVED** [review]
